### PR TITLE
Testing on travis-ci

### PR DESCRIPTION
**Pull request by MrBago**
This PR adds a travis config file and does a little refactoring to stay within travis's memory limit and pass tests using python3.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+sudo: required
```
Comment (MrBago) [inline .travis.yml]: This file is copied from graphframes.

File: python/tests/transformers/image_utils.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def compareArrays(self, values1, values2):
         """
         values1 & values2 are {key => numpy array}.
         """
-        for k, v1 in values1.iteritems():
+        for k, v1 in values1.items():
```
Comment (MrBago) [inline python/tests/transformers/image_utils.py]: For python3 compatibility.

File: python/tests/transformers/keras_image_test.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 from sparkdl.transformers.utils import InceptionV3Constants
 from ..tests import SparkDLTestCase
 from .image_utils import ImageNetOutputComparisonTestCase
-import image_utils
+from . import image_utils
```
Comment (MrBago) [inline python/tests/transformers/keras_image_test.py]: python3

File: python/tests/transformers/named_image_test.py [hunk 4]
```diff
@@ -34,12 +32,11 @@
 
 class NamedImageTransformerImagenetTest(SparkDLTestCase):
 
-    def test_inceptionV3_prediction(self):
```
Comment (MrBago) [inline python/tests/transformers/named_image_test.py]: Breaking this test up allows unused variables to go out of scope and be GCed.

**Comment by MrBago** [comment]
@sueann Travis up and running!

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (phi-dbq) [inline .travis.yml]: We might need to add 
```
jdk:
 - oraclejdk8 # openJDK crashes sometimes; tensorframes is compiled w.r.t JDK8
```
following TensorFrames Travis settings https://github.com/databricks/tensorframes/blob/master/.travis.yml

**Comment by phi-dbq** [comment]
LGTM!

**Comment by MrBago** [comment]
@sueann Can you take a look at this when you get a chance. I've rebased and resolved conflicts (with @phi-dbq's help).

File: python/tests/transformers/named_image_test.py [hunk 6]
```diff
@@ -34,27 +32,32 @@
...
         kerasPredict = model.predict(prepedImaged)
-
-        # test: _buildTfGraphForName
-        # Run the graph produced by _buildTfGraphForName and compare the result to above keras
-        # result.
+        # These values are used by multiple tests so cache them on class setup.
+        cls.imageArray = imageArray
+        cls.kerasPredict = kerasPredict
+
+    def test_buildtfgraphforname(self):
```
Comment (sueann) [inline python/tests/transformers/named_image_test.py]: camelCase

**Pull request merged**